### PR TITLE
Refactor safe extract method to fix issue 6215

### PR DESCRIPTION
**Pull request by liqunfu**
### Description
#6215

### CODE REVIEW

File: onnx/utils.py [hunk 1]
```diff
@@ -232,3 +233,65 @@ def extract_model(
...
+    of the tarfile module by checking for the availability of certain attributes
+    or methods before invoking them.
+
+    Parameters:
+    - model_tar_path (str): The path to the tar file to be extracted.
+    - local_model_with_data_dir_path (str): The directory path where the tar file
+      contents will be extracted to.
+
+    Returns:
+    None
```
Comment (justinchuby) [inline onnx/utils.py:283]:
```suggestion
    Args:
        model_tar_path: The path to the tar file to be extracted.
        local_model_with_data_dir_path: The directory path where the tar file
      contents will be extracted to.
```

nit

File: onnx/utils.py [hunk 2]
```diff
@@ -232,3 +233,65 @@ def extract_model(
...
+        elif member.issym() or member.islnk():
+            raise RuntimeError(
+                f"The tarball member {member_path} in downloading model contains "
+                f"symbolic links which may contain harmful payload."
+            )
+        result.append(member)
+    return result
+
+
+def extract_model_safe(model_tar_path: str, local_model_with_data_dir_path: str) -> None:
```
Comment (justinchuby) [inline onnx/utils.py:267]: Prefer supporting os.PathLike as well for any paths.
```suggestion
def extract_model_safe(model_tar_path: str | os.PathLike, local_model_with_data_dir_path: str | os.PathLike) -> None:
```

**Review by gramalingam — APPROVED** [review]

**Comment by gramalingam** [comment]
@liqunfu : should we merge this? I think renaming `extract_model_safe` to `_extract_model_safe` to make it private makes sense, as suggested above.

File: onnx/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (liqunfu) [reply onnx/utils.py]: renamed it to "_extract_model_safe"

**Pull request merged**